src/modules/tasks.js

```diff
@@ -1,39 +1,81 @@
-/* eslint-disable */
-
-import { forEach } from "lodash"
-
-let array = [
-  { name: "do homework", },
-  { name: "do workout", },
-  { name: "go to school", },
-  { name: "do homework", }
-];
+import Task from './task.js';
 
 export default class Tasks {
-  show(){
-    array.forEach(element => {
-      let container = document.querySelector('#list')
-    let div = document.createElement('div')
-    div.className = "d-flex justify-content-between px-4 align-items-center"
-    let check = document.createElement('input')
-    check.className=""
-    check.setAttribute('type','checkbox')
-    let h3 = document.createElement('h4')
-    h3.className = "m-0 col-10"
-    h3.textContent = `${element.name}`
-    let dots = document.createElement('i')
-    dots.className = "fa-solid fa-ellipsis-vertical"
-    let fragment = document.createDocumentFragment()
-    div.append(check,h3, dots)
-    fragment.append(div)
-    container.append(fragment)  
-    });
+  constructor() {
+    this.listTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+    this.show();
   }
-}
-
 
+  show() {
+    document.querySelector('#list').innerHTML = '';
+    this.listTasks.forEach((element) => {
+      const container = document.querySelector('#list');
+      // MAIN DIV
+      const div = document.createElement('div');
+      div.setAttribute('data-index', `${element.index}`);
+      div.className = 'd-flex justify-content-between px-4 align-items-center taskElement';
+      const check = document.createElement('input');
+      check.className = '';
+      check.setAttribute('type', 'checkbox');
+      const inputText = document.createElement('input');
+      inputText.setAttribute('type', 'text');
+      inputText.className = `m-0 col-10 border-0 input-txt ${element.index}`;
+      inputText.value = `${element.name}`;
+      const div2 = document.createElement('div');
+      const edit = `
+      <button type="button" class="fs-6 px-1 border-0 edit pe-2" data-bs-toggle="modal" data-index=${element.index}   data-bs-target="#exampleModal">
+      Edit
+      </button>
+      `;
+      div2.innerHTML = edit;
 
+      // trash icon
+      const dots = document.createElement('i');
+      dots.setAttribute('data-index', `${element.index}`);
+      dots.className = 'fa-solid fa-trash';
+      const fragment = document.createDocumentFragment();
+      div.append(check, inputText, div2, dots);
+      fragment.append(div);
+      container.append(fragment);
+    });
+  }
 
+  add() {
+    const taskText = document.querySelector('#task-input').value;
+    if (taskText === ' ') return;
+    const index = this.listTasks.length + 1;
+    const task = new Task(taskText, 'false', index);
+    this.listTasks.push(task);
+    localStorage.setItem('tasks', JSON.stringify(this.listTasks));
+    this.update();
+    this.show();
+  }
 
+  edit(index, newName) {
```

Comment:
- [ ] **[OPTIONAL] Good job so far on implementing a function for editing task descriptions** However, I think for the best user experience it would be better to replicate [the given demo video ](https://www.youtube.com/watch?v=AcUd-_Yjjqg)on editing descriptions 🙏